src/components/Icon/index.tsx

```diff
@@ -45,61 +45,60 @@ type IconProps = {
     WithThemeProps &
     WithStyleUtilsProps;
 
-// We must use a class component to create an animatable component with the Animated API
-// eslint-disable-next-line react/prefer-stateless-function
-class Icon extends PureComponent<IconProps> {
-    // eslint-disable-next-line react/static-property-placement
-    public static defaultProps = {
-        width: variables.iconSizeNormal,
-        height: variables.iconSizeNormal,
-        fill: undefined,
-        small: false,
-        inline: false,
-        additionalStyles: [],
-        hovered: false,
-        pressed: false,
-    };
-
-    render() {
-        const width = this.props.small ? variables.iconSizeSmall : this.props.width;
-        const height = this.props.small ? variables.iconSizeSmall : this.props.height;
-        const iconStyles = [this.props.StyleUtils.getWidthAndHeightStyle(width ?? 0, height), IconWrapperStyles, this.props.themeStyles.pAbsolute, this.props.additionalStyles];
-        const fill = this.props.fill ?? this.props.theme.icon;
-
-        if (this.props.inline) {
-            return (
-                <View
-                    testID={`${this.props.src.name} Icon`}
-                    style={[this.props.StyleUtils.getWidthAndHeightStyle(width ?? 0, height), this.props.themeStyles.bgTransparent, this.props.themeStyles.overflowVisible]}
-                >
-                    <View style={iconStyles}>
-                        <this.props.src
-                            width={width}
-                            height={height}
-                            fill={fill}
-                            hovered={this.props.hovered?.toString()}
-                            pressed={this.props.pressed?.toString()}
-                        />
-                    </View>
-                </View>
-            );
-        }
+const defaultProps = {
```

Comment:
Sorry for not seeing this earlier @rayane-djouah, but we shouldn't use `defaultProps` for TypeScript files – just provide a default when destructuring props:

```tsx
function Icon({src, width = variables.iconSizeNormal, ...}) {}
```

Reply:
Just qualifying this a bit, we _do_ still need to use `defaultProps` for class components. But not functional ones like this one.